File: Unit2-06/Java/Jet.java
/*
* This class creates a jet which is a type of airplane.
*
* @author  Kenny Le
* @version 1.0
* @since   2024-04-12
*/

/**
 * This is the class.
*/
public class Jet extends Airplane {
    /**
     * Sets the jet's speed multiplier.
    */
    private static int multiplier = 2;

    /**
     * Creates the base jet.
    */
    public Jet() {
    }

    /**
     * Sets the speed of the jet.
     *
     * @param newSpeed of the jet
    */
    public void setSpeed(int newSpeed) {
        super.setSpeed(newSpeed * Jet.multiplier);
    }

    /**
     * Accelerates the jet, changing speed.
    */
    public void accelerate() {
        super.setSpeed(this.getSpeed() * Jet.multiplier);
    }
}

File: Unit2-06/Typescript/Jet.ts
/**
*
* This class creates a jet which is a type of airplane
*
* By:      Evan Beaudoin
* Version: 1.0
* Since:   2024-04-12
*/

import { Airplane } from './Airplane'

export class Jet extends Airplane {
  // Creates the base vehicle
  public constructor() {
    super()
  }

  // Sets the speed of the jet
  public setSpeed(newSpeed: number) {
    super.setSpeed(newSpeed * 2)
  }

  // Accelerates the jet, changing speed
  public accelerate() {
    super.setSpeed(this.speed * 2)
  }
}
